python/ray/data/dataset.py

```diff
@@ -1147,6 +1149,11 @@ def to_tf(self,
                 of `tf.TypeSpec` objects corresponding to (features, label).
             feature_columns (Optional[List[str]]): List of columns in datasets
                 to use. If None, all columns will be used.
+            enable_autoshard (bool): Enables tensorflow built-in autosharding
+                (https://www.tensorflow.org/tutorials/distribute/input) for the
+                resulting TF Dataset. Defaults to False. It is recommended
+                to use ``split()`` to shard the Dataset instead of
+                auto-sharding.
```

Comment:
when would you want to do this?

Reply:
I don't have a particular use case in mind. Are you saying to remove this configurability? I feel like if we are going to modify the default behavior of TF Datasets, we should at least provide an option to change it back.

Reply:
I was just wondering if there's an actual situation for this arg to be useful. If we don't know of any, I think it makes more sense to keep this as an undocumented env var, and if users ask about it, we can 'elevate' it into a documented argument. 

It's just harder to go the other way.